### PR TITLE
Add recipe for mcdev

### PR DESCRIPTION
**Pull request by kotbaton**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

### CODE REVIEW

**Comment by marek-iiasa** [comment]
I confirm the change.

**Comment by marek-iiasa** [comment]
All ok

File: recipes/mcdev/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - setuptools >=61.0
+    - pip
+  run:
+    - python >=3.11,<3.12
+    - pyomo ==6.7.1
+    - pyyaml
+    - seaborn
+    - glpk
+    - dill
+    - numpy ==1.26.4,<2.0.0
```
Comment (xhochy) [inline recipes/mcdev/meta.yaml:31]: Do you really want an exact match here? Shouldn't it be the following?
```suggestion
    - numpy >=1.26.4,<2.0.0
```

Comment (kotbaton) [reply recipes/mcdev/meta.yaml]: Hi, thank you for the suggestion. I've improved the recipe according to it.

**Review by xhochy — APPROVED** [review]

**Pull request merged**